src/mpid/ch4/src/ch4_progress.h

```diff
@@ -14,21 +14,26 @@
 cvars:
     - name        : MPIR_CVAR_CH4_GLOBAL_PROGRESS
       category    : CH4
-      type        : boolean
-      default     : 1
+      type        : enum
+      default     : normal
       class       : none
       verbosity   : MPI_T_VERBOSITY_USER_BASIC
       scope       : MPI_T_SCOPE_LOCAL
-      description : >-
-        If on, poll global progress every once a while. With per-vci configuration, turning global progress off may improve the threading performance.
+      description : |-
+        Sets the frequency of polling global progress. It is only relevant when multiple VCIs are enabled.
+        low - only poll global progress very infrequently
+        normal - the default frequency for typical applications
+        high - always poll global progress
 
 === END_MPI_T_CVAR_INFO_BLOCK ===
 */
 
 /* Global progress (polling every vci) is required for correctness. Currently we adopt the
  * simple approach to do global progress every MPIDI_CH4_PROG_POLL_MASK.
  */
-#define MPIDI_CH4_PROG_POLL_MASK 0xff
+#define MPIDI_CH4_PROG_POLL_NORMAL_MASK 0xff    /* 256 x 1us = 1/4 ms */
+#define MPIDI_CH4_PROG_POLL_LOW_MASK 0xfffffff  /* 268435456 x 1us = 4.5 min. */
+#define MPIDI_CH4_PROG_POLL_HIGH_MASK 0x1       /* always global progress */
```

Comment:
It would probably be helpful for debugging and experiments to also provide another option to completely disable the global progress.

Reply:
I don't think there will be any effective difference between the LOW frequency and the OFF.

Reply:
One thing I'm not fully sure yet is: Could these infrequent global progress still have an ability to create a contention, for  applications or benchmarks that are known to not need a global progress.

Reply:
A contention every 4 minutes is not going to affect your benchmark numbers.